app/services/Worker.php

```diff
@@ -3,70 +3,83 @@
 
     /**
-     * @var Security
+     * @var WorkerValidator
      */
-    private $securityService;
+    private $workerValidator;
 
     /**
      * @var Workers
      */
     private $workerModel;
 
     /**
-     * Constructs the needed services, set in DI, for session, security and model
-     * @param SessionAdapter $sessionService
-     * @param Security $securityService
+     * Constructs the needed services, set in DI, for session, security, model,
+     * flashSession and view
+     * @param Authentication $authenticationService
+     * @param WorkerValidator $workerValidator
      * @param Workers $workerModel
+     * @param FlashSession $flashSession
+     * @param View $view
      */
     public function __construct(
-        SessionAdapter $sessionService,
-        Security $securityService,
-        Workers $workerModel
+        AuthenticationService $authenticationService,
+        WorkerValidator $workerValidator,
+        Workers $workerModel,
+        FlashSession $flashSession,
+        View $view
     ) {
-        $this->sessionService = $sessionService;
-        $this->securityService = $securityService;
+        parent::__construct($flashSession, $view);
+        $this->authenticationService = $authenticationService;
+        $this->workerValidator = $workerValidator;
         $this->workerModel = $workerModel;
     }
 
     /**
-     * Searches for the first match between the accessKey and the password of admin type
-     * workers from the database and if found, sets the worker's id in the session
-     * @param string $accessKey
-     * @return boolean
+     * If $worker array was submitted and validation is ok, registers the worker
+     * @param array $worker
+     * @return bool
      */
-    public function login(string $accessKey)
+    public function register(array $worker): bool
     {
-        $admins = $this->workerModel->find([
-            'admin = 1',
-        ]);
+        if (true === empty($worker)) {
+            return false;
+        }
 
-        foreach ($admins as $admin) {
-            if (true === $this->securityService->checkHash($accessKey, $admin->getPassword())) {
-                $this->sessionService->set('worker_session', $admin->getId());
+        $errors = $this->workerValidator->validation($worker);
+        $this->view->errors = $errors;
 
-                return true;
+        if (0 === count($errors)) {
+            if ('1' === $worker['admin']) {
+                $worker['password'] = $this->authenticationService->hashPassword($worker['password']);
             }
+
+            $this->workerModel->setLastName($worker['lastName']);
+            $this->workerModel->setFirstName($worker['firstName']);
+            $this->workerModel->setEmail($worker['email']);
+            $this->workerModel->setPassword($worker['password']);
+            $this->workerModel->setAdmin($worker['admin']);
+            $created = $this->workerModel->create();
+
+            if (true === $created) {
```

Comment:
this could had been refactored like this
if ( true === $this->workerModel->create()) {
flash session set
return true
}

if create() return bool.... you can return it directly